apps/pages/src/components/EditPage.vue

```diff
@@ -1,115 +1,278 @@
+          </div>
+          <div class="position-relative mt-4 shadow rounded">
+            <div
+              class="px-2 size-7 sticky-top d-flex justify-content-start bg-white"
+            >
+              <h2 class="flex-grow-1 h6 m-0 p-2">JS</h2>
+              <div class="d-flex">
+                <IconAction
+                  icon="file-code"
+                  tooltip="Format code"
+                  @click="formatEditor('javascript')"
+                />
+              </div>
+            </div>
+            <div ref="javascript" />
+          </div>
+          <div class="position-relative mt-4 shadow rounded">
+            <div
+              class="px-2 size-7 sticky-top d-flex justify-content-start bg-white"
+            >
+              <h2 class="flex-grow-1 h6 m-0 p-2">Resources</h2>
+            </div>
+            <form class="p-4">
+              <legend>Add external dependencies to your page</legend>
+              <div class="">
+                <label>CSS dependencies</label>
+                <ArrayInput
+                  id="css-urls"
+                  columnType="HYPERLINK_ARRAY"
+                  v-model="content.dependencies.css"
+                  description="Paste the URL to the CSS file (.css, min.css, etc.)"
+                />
+              </div>
+              <div>
+                <label for="javascript-urls">JavaScript dependencies</label>
+                <ArrayInput
+                  id="javascript-urls"
+                  columnType="HYPERLINK_ARRAY"
+                  v-model="content.dependencies.javascript"
+                  description="Paste the URL to the JS file (.js, min.js, etc.)"
+                />
+              </div>
+            </form>
+          </div>
+        </div>
+        <div class="position-relative col-5 bg-light shadow">
+          <div ref="pagePreview" class="sticky-top top-0 pt-2 pb-2 h-100"></div>
+        </div>
       </div>
     </div>
   </div>
 </template>
-
 <script>
 import {
+  InputText,
+  IconAction,
   ButtonAction,
-  ButtonAlt,
+  ButtonOutline,
   MessageError,
   MessageSuccess,
   Spinner,
+  ArrayInput,
 } from "molgenis-components";
 import { request } from "graphql-request";
-import { QuillEditor } from "@vueup/vue-quill";
-import "@vueup/vue-quill/dist/vue-quill.snow.css";
+import * as monaco from "monaco-editor";
+
+import { toRaw } from "vue";
+import { getPageSetting } from "../utils/getPageSetting";
+import { generateHtmlPreview } from "../utils/generateHtmlPreview";
+
+const editorLanguages = ["html", "css", "javascript"];
 
 export default {
   components: {
+    InputText,
     ButtonAction,
-    ButtonAlt,
+    ButtonOutline,
     MessageError,
     MessageSuccess,
     Spinner,
-    QuillEditor,
+    IconAction,
+    ArrayInput,
   },
   data() {
     return {
       graphqlError: null,
       success: null,
       loading: false,
-      draft: "<h1>New page</h1><p>Add your contents here</p>",
+      html: {},
```

Comment:
why both the seperate css and the content nested one?

Reply:
By separating the language content, it is easier to generate the HTML preview. CSS and JavaScript need to be processed and inserted into the document in a different way than HTML. Here's how JS is processed: https://github.com/molgenis/molgenis-emx2/blob/4a2836c68198060a000d6f8889846857b75573d9/apps/pages/src/utils/generateHtmlPreview.js#L20